File: src/main/java/ch/uzh/ifi/access/model/dto/CourseDTO.java
package ch.uzh.ifi.access.model.dto;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Data
@NoArgsConstructor
public class CourseDTO {
    public String slug;
    public String repository;
    public String logo;
    public Map<String,CourseInformationDTO> information = new HashMap<>();
    public String defaultVisibility;
    public String overrideVisibility;
    public LocalDateTime overrideStart;
    public LocalDateTime overrideEnd;
    public String studentRole;
    public String assistantRole;
    public String supervisorRole;
    public List<MemberDTO> supervisors = new ArrayList<>();
    public List<MemberDTO> assistants = new ArrayList<>();
    public List<String> assignments = new ArrayList<>();
}

File: src/main/kotlin/ch/uzh/ifi/access/model/dto/CourseDTO.kt
package ch.uzh.ifi.access.model.dto

import lombok.Data
import java.time.LocalDateTime

@Data
class CourseDTO (
    var slug: String? = null,
    var repository: String? = null,
    var logo: String? = null,
    var information: MutableMap<String, CourseInformationDTO> = HashMap(),
    var defaultVisibility: String? = null,
    var overrideVisibility: String? = null,
    var overrideStart: LocalDateTime? = null,
    var overrideEnd: LocalDateTime? = null,
    var studentRole: String? = null,
    var assistantRole: String? = null,
    var supervisorRole: String? = null,
    var supervisors: MutableList<MemberDTO> = ArrayList(),
    var assistants: MutableList<MemberDTO> = ArrayList(),
    var assignments: MutableList<String> = ArrayList()
)
